### DISCUSSION TITLE
Implementation of Sortable Component

### DISCUSSION THREAD

**Discussion by aruldd-twilio** [twilio-labs/paste#3221] [General] [answered]
We have a use case for Sortable Component, and we can see there is a "Drag and Drop Pattern" in the 12.12 release candidate. 

<img width="533" alt="image" src="https://user-images.githubusercontent.com/111948559/237018271-58021016-1df4-4fe5-9808-14f9cc89352c.png">

We can take this component up as a part of flex development and contribute back to paste. We need the following clarity for estimating the effort required. 

1. Can we do the implementation with a library [dnd-kit](https://docs.dndkit.com/presets/sortable/usesortable) or do we need to implement it from scratch? 
2. Is there an RFC we can use to implement this ?

**Comment by SiTaggart** [comment, marked as answer]
Hi @aruldd-twilio 

Glad to see you are interested in being an active part of Paste, unfortunately the Drag and Drop pattern is a lot less Engineering, and much more Design. The Pattern will more centered around design guidance, visual affordances, interaction design and accessibility. It won't actually ship a component.

It's pretty hard for us to ship a generic sortable component that is flexible to suit everyones needs. When you boil it down, it's no different to what a third party library would do for us at that point. Which is likely all we'd do, pick a library, wrap it, and export it.

The two libraries that we are going to recommend is [React DnD](https://react-dnd.github.io/react-dnd/about), coupled with this [React DnD Backend](https://github.com/discord/react-dnd-accessible-backend) built by Discord. That will give you the best, accessible drag and drop user interaction right now.

You can use these in conjunction with Paste Styling Primitives like Box and Text to create your custom component.

What you have in the screenshot is likely a little too product specific and not something we'd add to the Library long term.

Sorry that response probably isn't what you were expecting, but hopefully pointing you int he direction of some good libraries should get you going towards building your product feature.
